lib/pbench/server/api/resources/query_apis/__init__.py

```diff
@@ -335,30 +393,47 @@ def __init__(self, config: PbenchServerConfig, logger: Logger, schema: Schema):
         self.es_url = f"http://{host}:{port}"
         self.schema = schema
 
-    def _get_user_term(self, user: str) -> dict:
+    def _get_user_term(self, json: Dict[str, Any]) -> dict:
         """
-        _get_user_term Generate the user term for an Elasticsearch document
-        query. If the specified user parameter is not None, search for
-        documents with that value as the authorized owner. If the user
-        parameter is None, instead search for documents with public access.
+        Generate the user term for an Elasticsearch document query. If the
+        specified "user" parameter is not None, search for documents with that
+        value as the authorized owner. If the "access" parameter is "public"
+        or if the "user" parameter is missing (or None or ""), search for
+        published data.
+
+        TODO: Currently this code supports either public or private data, but
+        not both. The syntax here allows {"user": "me", "access": "public"}
+        to combine the terms; supporting combinations requires working out
+        the right Elasticsearch binary query syntax, and I'll plan to add that
+        in a separate PR.
+
+        TODO: does {"user": "me", "access": "public"} mean we return all data
+        owned by "me" *AND* all published data, or all data owned by "me" which
+        has been published? Do we want to support both modes? This may depend
+        on how we support "publishing" data; TBD.
 
         Args:
-            user: Pbench internal username or None
+            JSON query parameters containing:
+                "user": Pbench internal username or None
+                "access": Access category, "public" or "private"
 
         Returns:
             An assembled Elasticsearch query term
         """
+        user = json.get("user")
+        access = json.get("access", "private" if user else "public")
         if user:
             term = {"authorization.owner": user}
+        elif access == "private":
+            raise UnsupportedAccessMode(user, access)
         else:
             term = {"authorization.access": "public"}
         return term
```

Comment:
Yes, I considered centralizing the user/access consistency checking, because I agree this isn't a good place for the exception. The question is more of generality, since not all the ES wrapper APIs necessarily have both or either. I thought briefly of checking the subclass schema, which of course I could do, but I didn't remember to put that together.

E.g., it could be part of Schema validation, based on the presence of both user and access parameters... and I actually kinda like that idea. I think.

Reply:
I think you mistook my meaning:  I would prefer that no exception be raised at all, here or elsewhere!  😀  That is, _I_ think that a null user with either a "private" or null access is a legal request.  It might not be a _useful_ request, but it should not be illegal.  Null/null should produce all publicly-available data; Null/"private" should produce no data.  (I can see an argument for not bothering to query ES for the latter, but I don't think the former should result in an error...and I think that #2370 says as much as well.)